### ISSUE TITLE
Create Scopes & Collections - ensureCollections

### ISSUE THREAD

**Issue by AV25242** [couchbaselabs/node-ottoman#309]
In the current implementation, the `ottoman.start()` method is used to bootstrap Ottoman.

- _ottoman.start()_ method accepts configuration `useCollections()` this should be removed.

Create / modify  methods 

   **ensureIndexes** -  creates indexes depending on the scenario.
   **ensureCollections** - creates scopes and collections depending on the scenario
 
   **start**  - calls `ensureCollections()` first and then `ensureIndexes()`. 

  `ensureCollections()` - Below are different scenarios

- [ ]  **Scenario 1**: Scopes & Collections _**UNSUPPORTED**_, If a user configures `scopeName` and `collectionName` then `ensureCollections()` should attempt to create the scope and collection, the attempt to create a scope and collection should fail because this feature is not supported by Couchbase Server. The code should rely on the exception thrown by the underlying SDK (probably unsupported exception) and then return a proper error message to the user: **_This version of Couchbase Server does not support scopes and collections_**.

 **Note**:  There is no mechanism currently that will let you know if the Cluster you are working with supports Scopes & Collections or that will give you the server version back, you will have to rely on the underlying SDK error message for this.
At the time of creating this issue, an [open issue ](https://issues.couchbase.com/browse/JSCBC-812) in Node.js SDK exists that takes care of throwing the right exception back. Until this is resolved using a global configuration in your tests assuming **Supported vs Unsupported Platform** 

- [ ]  **Scenario 2**: Scopes & Collections _**UNSUPPORTED**_, If the user **DOES NOT** configure a `scopeName` and `collectionName` then `ensureCollections()` should NOT attempt to create the `default` scope and `default` collection.

- [ ]  **Scenario 3**: Scopes & Collections _**SUPPORTED**_, If user configures `scopeName` and `collectionName` then `ensureCollections()` should attempt to create the scope and collection.

- [ ]  **Scenario 4**: Scopes & Collections _**SUPPORTED**_, If the user **DOES NOT** configure a `scopeName` and `collectionName` then `ensureCollections()` should NOT attempt to create `default` scope and `default` collection, because *_default* a scope cannot be dropped. A *_default* collection can be dropped but it cannot be recreated (so no point in attempting to recreate)

as a general guideline 

   **ensureCollections** should be called before **ensureIndexes**.
   **Global Scope and Collection configuration** takes precedence  over **Model Level Scope and Collection**
            
The repository below exhibits in general test cases with a certain combination of global and Model Configuration for 
[7.0 Server](https://github.com/httpJunkie/ottoman-examples/tree/master/examples-7-zero)
and for [Legacy (Unsupported Database)](https://github.com/httpJunkie/ottoman-examples/tree/master/examples-legacy)

Note : the above repository doesn't necessarily give all permutations and combinations of global vs model level configuration that should follow the general guidelines as discussed above. However, the repository gives you examples of how to do global configuration and what to expect as an outcome.

**Comment by gsi-alejandro** [comment]
@AV25242 @httpJunkie 

In Ottoman every model will map to a collection, if the collection name isn't provided explicitly in the options the collectionName will be the same that the model name, so these 2 chunks of code are the same.
```
const options = {} // assumes default
const Airline = connection.model('Airline', schema, options)
```
is the same that:
```
const options = {collectionName: "Airline"} 
const Airline = connection.model('Airline', schema, options)
```
It's clear?

Then for Legacy and Modern defaults, we will find out, we have models only with a collectionName and we don't know if need to attempt to create collections or not.

The above exposed is the current behavior of Ottoman.
----------------------------------------------------------------------------------------------------------------------------------
Do we notice you are trying to avoid this by making the collectionName model's option required, in order to know if we must attempt to create the collection?
That's the idea?

**Comment by brett19** [comment]
The default should be '_default' rather than the model name.  There is another global-level config option which we will likely want in the future which provides the behaviour you described, but for compatibility reasons it cannot be the default.

**Comment by AV25242** [comment]
@gsi-alejandro  as per the meeting on 12/01/2020 it was decided that

- When no collection name is specified as  global configuration or at model level configuration options then the collection name will be the Model name, i.e `Model name will act as in implicit collection name.`

- When a Global or Model level collection name is specified then it overrides the Implicit Collection name.

-  All documents generated via Ottoman will have _type filed which will equal to the Model Name.

-  A refdoc Index document will be created in the same location as the original document.

-  A refdoc Index will always be a "binary document", this way its not pulled up by any N1QL query.

-  Bootstrap will happen via start method, however if the Indexes and/or scopes and collections exists no start method is needed, or in other words a save to the document assumes collections and indexes exist either manually created or via start method and should work seamlessly or throw an exception (for missing indexes and/or collection).

-  A default Index for _type field will be created as opposed to creating a Primary Key Index.

Final note  [Issue](https://issues.couchbase.com/browse/JSCBC-812) that Identifies Cluster support for Scopes and Collections is now resolved.

**Comment by httpJunkie** [comment]
Blocked by 
https://github.com/bwgjoseph/mongoose-vs-ottoman/issues/4

**Comment by AV25242** [comment]
Even though the issue on the customer side is still open, the latest comments indicates that these issues can be closed. We may open a new issue as needed
